src/wp-includes/block-supports/elements.php

```diff
@@ -31,90 +31,79 @@ function wp_get_elements_class_name( $block ) {
  * @return string Filtered block content.
  */
 function wp_render_elements_support( $block_content, $block ) {
-	if ( ! $block_content || empty( $block['attrs'] ) ) {
+	static $heading_elements = array( 'heading', 'h1', 'h2', 'h3', 'h4', 'h5', 'h6' );
```

Comment:
This static variable feels overkill. It is that expensive to create an array of values?

Reply:
What is the downside of using a `static` here?

Indeed I wouldn't have suggested using a static if @dmsnell wasn't already using it (simply because the cost of creating an array is so little), but despite that, creating an array on every function call is _definitely_ more expensive than only creating it once even if irrelevant in the grand scheme of things.

So IMO, unless there's a downside from using a `static`, I'm not sure I agree with removing it.

Reply:
Static variables use up memory. If this array is not used, like on a page that doesnt have headers, then this cache is point.

Reply:
@spacedmonkey See my updated reply in https://github.com/WordPress/wordpress-develop/pull/5411#discussion_r1348826278: Even if it the array was used, `static` doesn't appear to be faster one bit.

Something to keep in mind for the future, there doesn't seem to be _any_ performance benefit to this, so we should avoid it unless it has a functional benefit.